src/FondOfSpryker/Yves/GoogleTagManager/Plugin/FilterControllerEventHandler/EnhancedEcommerceFilterControllerEventHandlerPlugin.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManager\Plugin\FilterControllerEventHandler;
+
+use Spryker\Yves\Kernel\AbstractPlugin;
+use SprykerShop\Yves\ShopApplicationExtension\Dependency\Plugin\FilterControllerEventHandlerPluginInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+
+/**
+ * @method \FondOfSpryker\Yves\GoogleTagManager\GoogleTagManagerFactory getFactory()
+ * @method \FondOfSpryker\Yves\GoogleTagManager\GoogleTagManagerConfig getConfig()()
+ */
+class EnhancedEcommerceFilterControllerEventHandlerPlugin extends AbstractPlugin implements FilterControllerEventHandlerPluginInterface
+{
+    protected function checkForValidController(string $className): bool
+    {
+        foreach ($this->getConfig()->getListenToControllers() as $controller) {
+            if (\strpos($className, $controller) === false) {
```

Comment:
macht das Sinn?

Reply:
Was macht denn deiner Meinung nach keinen Sinn?

Reply:
protected function checkForValidController(string $className): bool
    {
        foreach ($this->getConfig()->getListenToControllers() as $controller) {
            if (\strpos($className, $controller) === false) {
                continue;
            }
                return true;
        }
        return false;
    }

protected function checkForValidController(string $className): bool
    {
        foreach ($this->getConfig()->getListenToControllers() as $controller) {
            if (\strpos($className, $controller) !== false) {
                return true;
            }
        }

        return false;
    }
hier reicht doch nur ein return true oder du nimmst nur die zweite if oder du baust es um auf single return

protected function checkForValidController(string $className): bool
    {
$return = false;
        foreach ($this->getConfig()->getListenToControllers() as $controller) {
             if (\strpos($className, $controller) !== false) {
                $return = true;
break;
            }
        }

        return $return;
    }